aten/src/ATen/native/Distance.cpp

```diff
@@ -75,4 +88,269 @@ Tensor cosine_similarity(const Tensor& x1, const Tensor& x2, int64_t dim, double
   return w12.div_((w1 * w2).clamp_min_(eps));
 }
 
+template <typename scalar_t>
+static void to_squareform_frame(
+    scalar_t *input_p, scalar_t *output_p, int64_t dimension, int64_t size)
```

Comment:
you are using terms in non-standard ways which makes this a little difficult to read.  dimension usually refers to an dimension index and size usually refers to the size of the dimension, but these are both referring to sizes here.  How about something like "square_dim_size" and "vector_size" or something?